app/models/chargeHistory/ChargeHistoryModel.scala

```diff
@@ -33,14 +33,14 @@ case class ChargeHistoryModel(taxYear: String,
                               reversalReason: String,
                               poaAdjustmentReason: Option[String]) {
 
-  val reasonCode: ReversalReason = {
+  val reasonCode: Either[Throwable, String] = {
     if (poaAdjustmentReason.isDefined)
-      AdjustmentReversalReason
+      Right(AdjustmentReversalReason.value)
     else
       reversalReason match {
-        case "amended return" => AmendedReturnReversalReason
-        case "Customer Request" => CustomerRequestReason
-        case _ => UnknownReversalReason
+        case "amended return" => Right(AmendedReturnReversalReason.value)
+        case "Customer Request" => Right(CustomerRequestReason.value)
+        case _ => Left(throw new Exception("Unknown reversal reason"))
```

Comment:
no need to throw exception, just a create one instead; I would be surprised if you can create a UT for this change;

Reply:
also should be ok to return Either[Throwable, ReversalReason] and consume .value where it is actually required, this way it should be easy to write required UTs;